### ISSUE TITLE
Promise not firing emit

### ISSUE THREAD

**Issue by cspinillo** [alexa/alexa-skills-kit-sdk-for-nodejs#271]
I've been working on this all day, tried every solution between here and stack overflow. Here is where I'm at.

My intent. It shouldn't matter, but this is an intent in a state other than the "initial" state.:

```
    'GetMondoPick':function(){
      const ref = this;
      FindMondoBottle(result =>{
        console.log("RESULT IS: "+result);
        ref.emit(":ask","Do i need to shoot myself?")
      });
    },
```

I call the following:
`
```
function FindMondoBottle(cb){
  var apiOptions = {
    host: API_ENDPOINT,
    path: API_PATH+"/products/club-items",
    method: 'GET',
    json: true
  }

  var apiRequest = http.request(apiOptions, (mondo) =>{
    console.log("Got the request atleast")
    var data = "";
    mondo.setEncoding('utf8');
    mondo.on('data', chunk => {
      data += chunk
    });
    mondo.on('end',()=>{
      console.log("Got to End")
      cb(data)
    })
  })
  apiRequest.end();
}
```
`
this.emit is just not firing, at all. If it is, it's out of scope and something isn't connecting along. I have tried request-promise. I've tried fat arrow. I've tried setting a reference (As above). I've tried calling Alexa directly. I can not, for the life of me, get the scoping right on this.

What am I missing here?

EDIT:

Outputting the "this" object keys, i see this:
```on,emit,emitWithState,handler,i18n,locale,t,event,attributes,context,callback,name,isOverridden,response ```

So, it is, in fact, there to call. Also, emitWithState works just fine.

**Comment by TianTXie** [comment]
@cspinillo ， An example skill has api call in handler function that you could refer is in [this link](https://github.com/alexa/alexa-cookbook/tree/master/handling-responses/progressive-response) . If you still need help, could you post the cloudwatch log in AWS console here. It would be helpful to figure out what is happen.

**Comment by cspinillo** [comment]
Thanks.

I'm going to have to do a massive Diff on git to figure out exactly what changed, but I ended up refactoring the whole thing and stepped away from fat arrows, for the time being, at least. I needed something reliable and went the promise route with request-then. This is working reliably for me now, so I won't be messing with fat arrows for time being. 

Scope management in the SDK has proven brittle for me after several skills.

Where I ended up is, below.

Define request object:

`
var request = require('request-promise').defaults({
  baseUrl:API_ENDPOINT,
  json: true,
  headers: {
    'User-Agent': 'Request-Promise'
  }
})
`

Then ... 

`
'GetMondoPick':function(){
      const self = this;
      remoteCall(this)
    },
`

And finally:

`
function remoteCall(scope){
  request("/api-path/club-items")
    .then(function (data) {
    console.log("DATA: "+data[0].name)
    scope.emit(":tell","Name Is: "+data[0].name)
  })
  .catch(function (err) {
      console.log("err: "+err)
      scope.emit(":tell","We're ****** again")
  });
}
`
